src/schema.rs

```diff
@@ -845,8 +918,67 @@ mod impls {
                     let func_name = receive_name.entrypoint_name();
                     Ok(json!({"contract": contract_name, "func": func_name}))
                 }
+                Type::ULeb128(constraint) => {
+                    let int = deserial_biguint(source, *constraint)?;
+                    Ok(Value::String(int.to_string()))
+                }
+                Type::ILeb128(constraint) => {
+                    let int = deserial_bigint(source, *constraint)?;
+                    Ok(Value::String(int.to_string()))
+                }
+                Type::ByteList(size_len) => {
+                    let len = deserial_length(source, *size_len)?;
+                    let mut string = String::new();
```

Comment:
Use `with_capacity`. I assume the input here is trusted? Otherwise use something like `safe_with_capacity`.

Reply:
I cannot find any documentation for a `safe_with_capacity` function or something like it.
Whould you mind sharing a link?

Reply:
We have one defined in either wasm-chain-integration, concordium-contracts-common, or crypto-common.

It is just with_capacity, it just has a maximum capacity it preallocates to avoid denial of service with malicious input.

Reply:
I use a safe with capacity now